certbot/cert_manager.py

```diff
@@ -139,6 +139,67 @@ def update_certs_for_domain_matches(candidate_lineage, rv):
 
     return _search_lineages(config, update_certs_for_domain_matches, (None, None))
 
+def cert_path_to_lineage(config):
+    """ If config.cert_path is defined, find an appropriate value for config.certname
+    by searching through available files in config.renewal_configs_dir, and finding
+    one with an appropriate value for 'fullchain'."""
+
+    # pylint: disable=unused-argument
+    def update_cert_name_for_cert_path_match(candidate_lineage, rv):
+        """ Return the lineagename or return None. """
+
+        if candidate_lineage.fullchain == config.cert_path[0]:
```

Comment:
Oh fun. So the documentation says to give it the path to `cert.pem`, but as long as you use `--cert-path`, any of these will work:

- path to `live/.../cert.pem`
- path to `live/.../fullchain.pem`
- path to `archive/.../cert<k>.pem`
- path to `archive/.../fullchain<k>.pem`

And `k` doesn't have to be the most recent `k`, either.

The first two are easy enough to test for, they'll be in the renewal conf file directly. For the second two, the file specified should live directly inside the archive directory specified in the renewal conf file.

Question: if someone's set it up so that multiple lineages use the same archive directory, are we already broken? If not, we have a Design Decision to make here...